[BUG] Assigning memoryview function to variable is causing stacktrace when shared utility is used

### Describe the bug

Compiling code below with `cython --shared=_cython reproducer.pyx` will result with following stacktrace:

```python
File "/home/dev/cython/.venv/bin/cython", line 10, in <module>
    sys.exit(setuptools_main())
             ~~~~~~~~~~~~~~~^^
  File "/home/dev/cython/Cython/Compiler/Main.py", line 822, in setuptools_main
    return main(command_line = 1)
  File "/home/dev/cython/Cython/Compiler/Main.py", line 850, in main
    result = compile(sources, options)
  File "/home/dev/cython/Cython/Compiler/Main.py", line 741, in compile
    return compile_multiple(source, options, cache)
  File "/home/dev/cython/Cython/Compiler/Main.py", line 700, in compile_multiple
    result = compile_single(source, options, full_module_name=options.module_name, cache=cache, context=context)
  File "/home/dev/cython/Cython/Compiler/Main.py", line 671, in compile_single
    return run_pipeline(source, options, full_module_name, context)
  File "/home/dev/cython/Cython/Compiler/Main.py", line 567, in run_pipeline
    err, enddata = Pipeline.run_pipeline(pipeline, source)
                   ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/home/dev/cython/Cython/Compiler/Pipeline.py", line 418, in run_pipeline
    data = run(phase, data)
  File "/home/dev/cython/Cython/Compiler/Pipeline.py", line 400, in run
    return phase(data)
  File "/home/dev/cython/Cython/Compiler/Pipeline.py", line 50, in generate_pyx_code_stage
    module_node.process_implementation(options, result)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/dev/cython/Cython/Compiler/ModuleNode.py", line 315, in process_implementation
    self.generate_c_code(env, options, result)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dev/cython/Cython/Compiler/ModuleNode.py", line 647, in generate_c_code
    self.generate_declarations_for_modules(env, modules, globalstate)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dev/cython/Cython/Compiler/ModuleNode.py", line 863, in generate_declarations_for_modules
    vtab_list, vtabslot_list = self.sort_type_hierarchy(modules, env)
                               ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "/home/dev/cython/Cython/Compiler/ModuleNode.py", line 798, in sort_type_hierarchy
    assert key not in vtabslot_dict, key
           ^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: __pyx_array_obj
```

### Code to reproduce the behaviour:

```cython
cdef void memoryview_func_a(double [:] x):
    x[0] = 1

def test_memview_wrapping():
    cdef a = memoryview_func_a
```


### Expected behaviour

_No response_

### OS

Linux

### Python version

3.14

### Cython version

master

### Additional context

_No response_

--- a/Cython/Compiler/TreeFragment.py
+++ b/Cython/Compiler/TreeFragment.py
@@ -22,12 +22,12 @@ from . import UtilNodes
 
 
 class StringParseContext(Main.Context):
-    def __init__(self, name, include_directories=None, compiler_directives=None, cpp=False):
+    def __init__(self, name, include_directories=None, compiler_directives=None, cpp=False, options=None):
         if include_directories is None:
             include_directories = []
         if compiler_directives is None:
             compiler_directives = {}
-        Main.Context.__init__(self, include_directories, compiler_directives, cpp=cpp, language_level='3')
+        Main.Context.__init__(self, include_directories, compiler_directives, cpp=cpp, language_level='3', options=options)
         self.module_name = name
 
     def find_module(self, module_name, from_module=None, pos=None, need_pxd=1, absolute_fallback=True, relative_import=False):

--- a/Cython/Compiler/UtilityCode.py
+++ b/Cython/Compiler/UtilityCode.py
@@ -39,10 +39,6 @@ class NonManglingModuleScope(Symtab.ModuleScope):
 class CythonUtilityCodeContext(StringParseContext):
     scope = None
 
-    @property
-    def shared_utility_qualified_name(self):
-        return self.cython_scope.context.shared_utility_qualified_name if self.cython_scope else None
-
     def find_module(self, module_name, from_module=None, pos=None, need_pxd=True, absolute_fallback=True, relative_import=False):
         if from_module:
             raise AssertionError("Relative imports not supported in utility code.")
@@ -130,7 +126,8 @@ class CythonUtilityCode(Code.UtilityCodeBase):
         from . import Pipeline, ParseTreeTransforms
         context = CythonUtilityCodeContext(
             self.name, compiler_directives=self.compiler_directives,
-            cpp=cython_scope.is_cpp() if cython_scope else False)
+            cpp=cython_scope.is_cpp() if cython_scope else False,
+            options=cython_scope.context.options if cython_scope else None)
         context.prefix = self.prefix
         context.cython_scope = cython_scope
         #context = StringParseContext(self.name)

